xonsh/tools.py

```diff
@@ -1582,9 +1582,13 @@ def expanduser_abs_path(inp):
 def expand_case_matching(s):
```

Comment:
i know regex can be a little overwhelming, but they really pay off imho. it's easier to debug than the spaghetti per character parsing and much cleaner. also the very cool http://pythex.org/ to test your patterns!